### ISSUE TITLE
About detecting many extra boxes

### ISSUE THREAD

**Issue by pengjizhi** [yqyao/SSD_Pytorch#36]
Hello author, thank you for your hard work!
Drawing on your ideas, I am developing my own single-stage object detection network, but there are a lot of extra boxes in the visualization results.
Since my network structure is deep, I guess it may be caused by the disappearance of the gradient or the explosion of the gradient, but why do these two situations lead to the appearance of a large number of abnormal boxes, can you give any suggestions?
![lindau_000000_000019_detection](https://user-images.githubusercontent.com/40751283/70404437-083e0700-1a75-11ea-9fb8-dffdf4d54c1c.png)

**Comment by pengjizhi** [comment]
In addition, I don't think this is the cause of nms. In fact, such results are already the results after nms.

**Comment by yqyao** [comment]
How about your training loss ? @pengjizhi I think the value of your training loss may be very large.

**Comment by pengjizhi** [comment]
@yqyao 
Hello!
I also feel very strange about this, because in my display, the value of loss has been falling normally, so I didn't realize that this happened, and it wasn't until I tested it that I found a lot of exception boxes.

**Comment by yqyao** [comment]
What dataset you use for training ? @pengjizhi

**Comment by yin19961202** [comment]
Have you solved this problem?I've met the same issue. The false positive rate was very high. @pengjizhi

**Comment by pengjizhi** [comment]
@yin19961202  
Hello, I haven't found the main cause of this problem at present, I guess it may be because the model is too complicated, and it caused gradient explosion or gradient disappearance during training.
Currently I can solve this problem by reducing the learning rate.

**Comment by yin19961202** [comment]
Hello, I found that if I increased the threshold of the positive box (for example, 0.8), the extra boxes would be removed but some correct boxes would be removed, too. I also found the training loss was very high in the end. Can you share the setting of your learning rate? Thanks




| |
尹锐红
邮箱：13261061903@163.com
|

签名由 网易邮箱大师 定制

On 12/23/2019 09:34, pengjizhi wrote:

@yin19961202
Hello, I haven't found the main cause of this problem at present, I guess it may be because the model is too complicated, and it caused gradient explosion or gradient disappearance during training.
Currently I can solve this problem by reducing the learning rate.
